distributed/security.py

```diff
@@ -123,7 +123,8 @@ def get_connection_args(self, role):
         """
         d = {}
         tls = self.get_tls_config_for_role(role)
-        d['ssl_context'] = self._get_tls_context(tls, ssl.Purpose.SERVER_AUTH)
+        purpose = ssl.Purpose.SERVER_AUTH if hasattr(ssl, "Purpose") else None
+        d['ssl_context'] = self._get_tls_context(tls, purpose)
```

Comment:
Passing a string would only be important if we actually want to create a context, right? But then we are into the problem of re-implementing `create_default_context`. In other words: If there is no `Purpose`, any attempt to create a context _should_ fail, right?